src/components/__tests__/container.test.js

```diff
@@ -194,6 +195,23 @@ describe('Container', () => {
       expect(defaultRegistry.deleteStore).not.toHaveBeenCalled();
     });
 
+    it('should not cleanup from global if same type gets recreated meanwhile', async () => {
+      const Subscriber = createSubscriber(Store);
+      const renderPropChildren = jest.fn().mockReturnValue(null);
+      const children = <Subscriber>{renderPropChildren}</Subscriber>;
+      const { unmount } = render(<Container scope="s1">{children}</Container>);
+      unmount();
+
+      // given cleanup is scheduled, we fake the eventuality that the store gets replaced in meantime
+      defaultRegistry.getStore.mockReturnValue({
```

Comment:
so this is to _support_ `prevStoreState === storeState`, but it's sooooo synthetic.

Can the same example be recreated by flipping `scope` on sibling components, so they will change it in one tick?

Reply:
Tried that in multiple ways, but cannot replicate in test environment. React has lots of special cases around how `useEffect` are triggered and could not find a way to have the cleanup happen between render and `useEffect` in tests 😞 
One alternative is to mock `useExternalStore`, which feels as dirty as this test thought

Reply:
🤗 I feel you. The condition you want to represent synthetically seems to be equal to the reported problem. It will be good to have a "real use case", but if it's too hard to reproduce in _another environment_ - one can always verify manually, and if the problem is solved, then the test is doing the right thing and will protect behavior from future regressions.